### DISCUSSION TITLE
Can the Perception Module be used for Federates and how can it be extended to include other detections like pedestrians

### DISCUSSION THREAD

**Discussion by paulbourelly999** [eclipse/mosaic#338] [Q&A] [answered]
In the documentation this Perception Module is shown to be a Module used with a Mosiac Vehicle Application. Was wondering if this functionality can appropriately be reused for Application Ambassadors (Federates) to simulate object detection from sensors (multiple) and their output. For example if I am creating an Ambassador for some ITS (Intelligent Transportation Software) deployed at the roadside so I can test it in simulation with simulators like SUMO. Is there a way to reuse this Perception Module to spawn multiple simulated sensors and push detected object information to my ITS roadside software?

**Comment by schwepmo** [comment]
If I understand you correctly you want to deploy some form of static perception sensor, right? This currently isn't supported but shouldn't be too hard to implement.
The simplest way I can imagine would be to extend existing RSUs with a perception module. Since RSUs have a position and communication capabilities your goal should be possible.

You could then configure the apps running on your RSUs to send interactions to the RTI that will be handled by your additional ITS-ambassador, OR you just handle your ITS logic within the app of the RSU or an additional centralized server.

**Comment by paulbourelly999** [comment]
In this case I would only be able to create a single static perception sensor per RSU, right. Also to retrieve/trigger interactions for the detected objects I would need applications for each sensor/rsu. Just want to confirm I am understanding this correctly?

**Reply by schwepmo** [reply, marked as answer]
Yes, though using the `mapping_config.json` you could write one `SensingRSUApp` which can be the same for all deployed RSUs. Exemplary, you could then add central server which collects the data using the Cell-Simulator and deploy your ITS-software on there.

Having multiple sensors per RSU is also not unthinkable, however, we currently don't have the capacities to implement either feature. If you want to give it a try we could gladly support you along the way.

**Reply by paulbourelly999** [reply]
Thanks for the help. I think I understand now. For our functionality we ended up going a different way and integrating CARLA Sensors with CDASim. We have an ambassador for CARLA and we plan to create interactions to register a sensor and interactions to publish detections from the sensor. I saw on your website that you are currently working on CARLA Integration for the next release. CARLA Sensors may be something helpful to integrate as well if you are not planning to yet.
